src/main/java/com/netspi/awssigner/controller/TextComponentFocusListener.java

```diff
@@ -9,6 +9,12 @@
 import javax.swing.event.DocumentEvent;
 import javax.swing.event.DocumentListener;
 import javax.swing.text.JTextComponent;
+import javax.swing.undo.UndoManager;
+import javax.swing.event.UndoableEditEvent;
+import javax.swing.event.UndoableEditListener;
+import javax.swing.AbstractAction;
+import java.awt.event.ActionEvent;
+import javax.swing.KeyStroke;
 
 class TextComponentFocusListener<T extends Profile> implements FocusListener {
```

Comment:
This is somewhat related to the comment below, but is this class the appropriate place to add this functionality. 

This `TextComponentFocusListener` class is used across a lot of different text fields, not just the session policy text field. You can search across AWSSignerController for more examples, but here are just a few: https://github.com/NetSPI/AWSSigner/blob/0c2cace5cb44da31f060900f78d9d501e0060d5b/src/main/java/com/netspi/awssigner/controller/AWSSignerController.java#L251-L258

If I'm understanding the PR correctly, we're intending to add these functions to the Session Policy text field, but I think this will apply the functionality to lots of different text fields in the application. Can we investigate that and confirm? I might just be missing something.

Reply:
This is correct, all text fields utilizing this listener will now have undo/redo capability. I haven't seen any issues with this during the testing, each text field has its own history and even after switching from text field to text field, the history is working as expected.